src/Pagination/IterablePaginationInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace JiraRestApi\Pagination;
+
+use Exception;
+use Generator;
+
+/**
+ * The interface providing a way to iterate through all pages of a paginated query.
+ * @template T
+ */
+interface IterablePaginationInterface
+{
+    /**
+     * Returns all pages.
+     * @return Generator<PaginatedQueryResultInterface<T>>
```

Comment:
I think our phpcs or psalm complain about that, so I usually do it like this.

Also here it makes it clear that it will not load all 100500 pages at once.

Reply:
IMO the amount of pages it loads at once could be subject to change, and is an implementation detail. `iterable` includes `Traversable`, and `Generator` implements `Traversable` through `Iterator`. Like this, it is totally compatible and abstract. And if in your test you would need to provide a hard-coded list of `PaginatedQueryResultInterface`, you can :)

Reply:
:heavy_check_mark: